### PR TITLE
Add bond.rho from bond demo to rholang/examples

### PR DESCRIPTION
**Pull request by KentShikama**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

The title says it all

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: rholang/examples/bond/bond.rho [hunk 1]
```diff
@@ -0,0 +1,8 @@
+new retCh, PoSCh, rl(`rho:registry:lookup`), stdout(`rho:io:stdout`) in {
+  stdout!("About to lookup pos contract") |
+  rl!(`rho:rchain:pos`, *PoSCh) |
+  for(@(_, PoS) <- PoSCh) {
+    stdout!("About to bond") |
+    @PoS!("bond", 100, *retCh)
```
Comment (ArturGajowy) [inline rholang/examples/bond/bond.rho]: One more thing to point out during the demo is that this contract is going to accept a `deployerId` in the future. In it's current form, this relies on the (generalized) 'drain vault' vulnerability

**Review by ArturGajowy — APPROVED** [review]

File: rholang/examples/bond/bond.rho (hunk 1 above)
Comment (KentShikama) [reply rholang/examples/bond/bond.rho]: Noted

Comment (odeac) [inline rholang/examples/bond/bond.rho]: You should wait for the bond result on retCh and display a "bond successful message" or a "bond failed" message

**Review by odeac — CHANGES_REQUESTED** [review]
I would also check the bond result

**Review by JosephDenman — APPROVED** [review]

File: rholang/examples/bond/bond.rho (hunk 1 above)
Comment (KentShikama) [reply rholang/examples/bond/bond.rho]: PTAL

**Review by odeac — APPROVED** [review]

**Review by odeac — APPROVED** [review]

**Pull request merged**